### ISSUE TITLE
Can't Reproduce Results With tf.keras.layers.Conv2D

### ISSUE THREAD

**Issue by jsl303** [tensorflow/tensorflow#28415]
Putting the code below in the beginning of my script, I can consistently reproduce the result 100%. However this is only true if I only use Dense layer.

```
import numpy as np
import random as rn
import tensorflow as tf
import os
os.environ['PYTHONHASHSEED'] = '0'
np.random.seed(1)
rn.seed(2)
session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
from tensorflow.keras import backend as K
tf.set_random_seed(3)
sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
K.set_session(sess)
```

If I insert this one line "model.add(Conv2D(32, 3, activation='relu'))" before "model.add(Flatten())", it produces different results every time I rerun.

Input> flatten > dense produces consistent result, but input > conv2d > flatten > dense produces different result every time I run the code.

It looks like a bug to me, but I apologize if isn't.

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution: Debian GNU/Linux 9.8 (stretch) (GNU/Linux 4.9.0-8-amd64 x86_64)
- TensorFlow installed from (source or binary): conda install -c anaconda tensorflow-gpu
- TensorFlow version (use command below): 1.13.1
- Python version: 3.7.1
- CUDA/cuDNN version: 10.0
- GPU model and memory: Tesla P4 7611MiB

**Comment by muddham** [comment]
@jsl303  In order to expedite the trouble-shooting process, please provide whole code snippet to reproduce the issue reported here. Thanks!

**Comment by jsl303** [comment]
The testing code is at the bottom. Here are the results for 10 runs.
What's interesting is that the following runs match in the model with conv layers. 
Run 1 and 5, Run 2 and 3, run 7 and 8, Run 6 and 10

```
Run 1: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9909999966621399                                                                                                                
Run 2: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.991100013256073                                                                                                                                                                                                                                    
Run 3: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.991100013256073                                                                                                                  
Run 4: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9918000102043152                                                                                                                
Run 5: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9909999966621399                                                                                                                
Run 6: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9907000064849854                                                                                                                
Run 7: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9912999868392944                                                                                                                
Run 8: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9912999868392944                                                                                                                
Run 9: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9904999732971191                                                                                                                
Run 10: Dense accuracy: 0.9431999921798706, Conv accuracy: 0.9907000064849854                                                                                                                

```
Here's the code.

```
import numpy as np
import random as rn
import tensorflow as tf
import os
os.environ['PYTHONHASHSEED'] = '0'
np.random.seed(1)
rn.seed(2)
session_conf = tf.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
from tensorflow.keras import backend as K
tf.set_random_seed(3)
sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
K.set_session(sess)

from tensorflow.keras.datasets import mnist
from tensorflow.keras.layers import Dense, Dropout, Flatten, Conv2D, MaxPooling2D, Activation, Input
from tensorflow.keras.models import Model
from tensorflow.keras.utils import to_categorical

def dense_model(input_shape, output_shape):
	inputs = Input(input_shape)
	x = Flatten()(inputs)
	x = Dense(300, activation="relu")(x)
	x = Dense(100, activation="relu")(x)
	outputs = Dense(output_shape, activation='softmax')(x)
	model = Model(inputs=inputs, outputs=outputs)
	model.compile(loss="categorical_crossentropy", optimizer="sgd", metrics=["accuracy"])
	return model

def conv_model(input_shape, output_shape):
	inputs = Input(input_shape)
	x = Conv2D(32, 3, activation='relu')(inputs)
	x = Conv2D(64, 3, activation='relu')(x)
	x = MaxPooling2D(2)(x)
	x = Dropout(0.25)(x)
	x = Flatten()(x)
	x = Dense(128, activation='relu')(x)
	x = Dropout(0.5)(x)
	outputs = Dense(output_shape, activation='softmax')(x)
	model = Model(inputs=inputs, outputs=outputs)
	model.compile(loss="categorical_crossentropy", optimizer="adadelta", metrics=['accuracy'])
	return model

batch_size = 128
output_shape = 10
epochs = 12
img_rows, img_cols = 28, 28
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
input_shape = (img_rows, img_cols, 1)
x_train = x_train.astype('float32')/255
x_test = x_test.astype('float32')/255
y_train = to_categorical(y_train, output_shape)
y_test = to_categorical(y_test, output_shape)
print("Testing model with dense layers...")
model = dense_model(input_shape, output_shape)
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, verbose=1, validation_data=(x_test, y_test))
dense_score = model.evaluate(x_test, y_test, verbose=0)
print("Testing model with conv layers...")
model = conv_model(input_shape, output_shape)
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, verbose=1, validation_data=(x_test, y_test))
conv_score = model.evaluate(x_test, y_test, verbose=0)
print(f"Dense accuracy: {dense_score[1]}, Conv accuracy: {conv_score[1]}")


```

**Comment by muddham** [comment]
@jsl303 After running the code with epoch=1 got these results.
Dense accuracy: 0.5679987038135529, Conv accuracy: 0.06941995525737293
Dense accuracy: 0.5325789436578751, Conv accuracy: 0.06686571390144527
Dense accuracy: 0.5519545083522797, Conv accuracy: 0.06901443190686404
Dense accuracy: 0.547787981891632, Conv accuracy: 0.06690254995841532
Dense accuracy: 0.5438883091449738, Conv accuracy: 0.09017947126738728

For epoch=12 these are the below results.

Dense accuracy: 0.19799944971501826, Conv accuracy: 0.6604872384548187
 0.1981875723950565, Conv accuracy: 0.6766708342552185

As this is the custom code, This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!

**Comment by muddham** [comment]
@jsl303 Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
